presto-spi/src/main/java/io/prestosql/spi/block/ShortArrayBlockBuilder.java

```diff
@@ -102,7 +102,7 @@ public Block build()
         if (!hasNonNullValue) {
             return new RunLengthEncodedBlock(NULL_VALUE_BLOCK, positionCount);
         }
-        return new ShortArrayBlock(0, positionCount, valueIsNull, values);
+        return new ShortArrayBlock(0, positionCount, hasNullValue ? valueIsNull : null, values);
```

Comment:
```suggestion
        return new ShortArrayBlock(positionCount, hasNullValue ? Optional.of(valueIsNull) : Optional.empty(), values);
```

?

Reply:
No. 

Current is change is coherent with every other implementation of `buld()` method. If we decide to go with `Optional` I would prefer to change all of them at once.

Also using `Optional` in `Block` implementation has to be conscious decision. As far I understand that could have a negative impact on performance.

Reply:
> Also using Optional in Block implementation has to be conscious decision. As far I understand that could have a negative impact on performance.


The constructor accepting optional already exists (my proposal compiles).

> Current is change is coherent with every other implementation of buld() method.

Thanks for the context.

Then leave as is.